utils/ticker.go

```diff
@@ -0,0 +1,44 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package utils
+
+import (
+	"sync"
+	"time"
+)
+
+// Ticker is a timer that can be subscribed to. When the timer ticks,
+// all subscribers will receive a signal on the channel they were given
+// when subscribing.
+type Ticker struct {
+	interval      time.Duration
+	subscriptions []chan struct{}
+	lock          *sync.Mutex
+}
+
+func NewTicker(writeIntervalSeconds uint64) *Ticker {
+	return &Ticker{
+		interval: time.Duration(writeIntervalSeconds) * time.Second,
```

Comment:
```suggestion
func NewTicker(intervalSeconds uint64) *Ticker {
	return &Ticker{
		interval: time.Duration(intervalSeconds) * time.Second,
```

Nit: now that this in `utils` the interval isn't necessarily tied to writes to the DB